src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/ReviewCodeForSqlInjectionVulnerabilitiesTests.cs

```diff
@@ -3711,6 +3711,50 @@ public MyController(string x)
                 GetCSharpResultAt(9, 9, 7, 29, "SqlCommand.SqlCommand(string cmdText)", "void MyController.DoSomething(string input)", "string input", "void MyController.DoSomething(string input)"));
         }
 
+        [Theory]
+        [InlineData("st.Append(id);")]
+        [InlineData("st.AppendFormat(id, \"1\");")]
+        [InlineData("st.AppendFormat(\"{0}\", id);")]
+        [InlineData("st.AppendFormat(\"{0}{1}\", \"\", id);")]
+        [InlineData("st.AppendFormat(\"{0}{1}{2}\", \"\", \"\", id);")]
+        [InlineData("st.AppendFormat(\"{0}{1}{2}{3}\", \"\", \"\", \"\", id);")]
+        [InlineData("st.AppendFormat(\"{0}{1}{2}{3}{4}\", \"\", \"\", \"\", \"\", id);")]
+        [InlineData("st.AppendLine(id);")]
+        [InlineData("st.Append(id); st.CopyTo(0, arr, 0, 10)", true, "new string(arr)")]
+        [InlineData("st.Insert(0, id);")]
+        [InlineData("st.Replace(\"\", id);")]
+        [InlineData("st.Append(id); st.Clear()", false)]
```

Comment:
Should be possible, if:

- [ ] Sanitizing instance properties enumerable is added to `AddSanitizerInfo`
- [ ] Instead of just comparing the method/property name similar approach to `MethodMatcher` is implemented

However I've quickly looked in debugger what are the values in `CopyTo` method matcher and for some reason `ConstantValue` for the fourth argument is not specified. Need to enable some kind of analysis?

Sounds like a separate issue can be created for this feature.

Reply:
> However I've quickly looked in debugger what are the values in `CopyTo` method matcher and for some reason `ConstantValue` for the fourth argument is not specified. Need to enable some kind of analysis?

If you're looking at an `IArgumentOperation`, I think you need to look at its `.Value.ConstantValue`.

Reply:
Ah, yes, indeed.